File: bench/cross-language/json_parse/json_parse.ts
function main(): void {
  // Build an object with 10000 entries
  const data: Record<string, string> = {};
  for (let i = 0; i < 10000; i++) {
    data[`key_${i}`] = `value_${i}`;
  }

  // Serialize to JSON
  const jsonStr = JSON.stringify(data);

  // Parse back
  const parsed = JSON.parse(jsonStr) as Record<string, string>;

  // Access a field
  const found = parsed["key_9999"];
  console.log(`Found: ${found}`);
  console.log(`Count: ${Object.keys(parsed).length}`);
}

main();

File: bench/cross-language/json_parse/json_parse.py
import json


def main():
    # Build a dict with 10000 entries
    data = {}
    for i in range(10000):
        data[f"key_{i}"] = f"value_{i}"

    # Serialize to JSON
    json_str = json.dumps(data)

    # Parse back
    parsed = json.loads(json_str)

    # Access a field
    found = parsed["key_9999"]
    print(f"Found: {found}")
    print(f"Count: {len(parsed)}")


if __name__ == "__main__":
    main()
